gentleman/src/utils/userReducer.js

```diff
@@ -1,49 +1,81 @@
 const userReducer = (state, action) => {
   switch (action.type) {
+    case "UPDATE_CART":
+      return { ...state, cart: [...action.payload] };
+    case "UPDATE_WISHLIST":
+      return { ...state, wishlist: [...action.payload] };
     case "ADD_TO_CART":
       return {
         ...state,
-        cart: [...state.cart, { product: action.payload, quantity: 1 }],
+        cart: [...state.cart, action.payload],
       };
+
     case "ADD_TO_WISHLIST":
       if (
         state.wishlist.find((item) => item.id === action.payload.id) ===
         undefined
       )
         return { ...state, wishlist: [...state.wishlist, action.payload] };
       else return { ...state };
-    case "REMOVE_FROM_CART":
-      return {
-        ...state,
-        cart: state.cart.filter(
-          (item) => item.product.id !== action.payload.product.id
-        ),
-      };
-    case "REMOVE_FROM_WISHLIST":
+    // case "REMOVE_FROM_CART":
+    //   return {
+    //     ...state,
+    //     cart: [...action.payload],
+    //     // state.cart.filter( (item) => item.product.id !== action.payload.product.id),
+    //   };
+    // case "REMOVE_FROM_WISHLIST":
+    //   return {
+    //     ...state,
+    //     wishlist: [...action.payload],
+    //     // state.wishlist.filter((item) => item.id != action.payload.id)
+    //   };
+
+    // case "INCREASE_QUANTITY":
+    //   return {
+    //     ...state,
+    //     cart: [...action.payload],
+    //     // cart: state.cart.map((item) => {
+    //     //   return item.product.id === action.payload
+    //     //     ? { ...item, quantity: item.quantity + 1 }
+    //     //     : item;
+    //     // }),
+    //   };
+    // case "DECREASE_QUANTITY":
+    //   return {
+    //     ...state,
+    //     cart: [...action.payload],
+    //     // cart: state.cart.map((item) => {
+    //     //   return item.product.id === action.payload && item.quantity > 1
+    //     //     ? { ...item, quantity: item.quantity - 1 }
+    //     //     : item;
+    //     // }),
+    //   };
```

Comment:
Can remove these comments because you are doing the actions in UPDATE_CART and UPDATE_WISHLIST itself.

Reply:
yes kept them for future use , will integrate it with a real backend